invertedai/large/initialize.py

```diff
@@ -0,0 +1,296 @@
+            zip(existing_agent_states,existing_agent_attributes)
+        ))
+
+        conditional_agent_states = [x[0] for x in conditional_agents]
+        conditional_agent_attributes = [x[1] for x in conditional_agents]
+
+        region_predefined_agent_states = [] if region.agent_states is None else region.agent_states
+        all_agent_states = conditional_agent_states + region_predefined_agent_states
+        all_agent_attributes = conditional_agent_attributes + region.agent_attributes
+
+        try:
+            response = iai.initialize(
+                location=location,
+                states_history=None if len(all_agent_states) == 0 else [all_agent_states],
+                agent_attributes=[] if len(all_agent_attributes) == 0 else all_agent_attributes,
+                get_infractions=get_infractions,
+                traffic_light_state_history=traffic_light_state_history,
+                location_of_interest=(region_center.x, region_center.y),
+                random_seed=random_seed,
+                get_birdview=get_birdview,
+            )
+
+            if traffic_light_state_history is None and response.traffic_lights_states is not None:
+                traffic_light_state_history = [response.traffic_lights_states]
+
+        except InvertedAIError as e:
+            iai.logger.warning(e)
+            continue
+
+        # Remove all predefined agents before filtering at the edges
+        response_agent_states_sampled = response.agent_states[len(all_agent_states):]
+        response_agent_attributes_sampled = response.agent_attributes[len(all_agent_states):]
+        response_recurrent_states_sampled = response.recurrent_states[len(all_agent_states):]
+
+        # Filter out agents that are not inside the ROI to avoid collision with other agents not passed as conditional
+        # SLACK is for removing the agents that are very close to the boundary and they may collide with agents not 
+        # labelled as conditional
+        valid_agents = list(filter(
+            lambda x: inside_fov(center=region_center, agent_scope_fov=region_size - SLACK, point=x[0].center),
+            zip(response_agent_states_sampled, response_agent_attributes_sampled, response_recurrent_states_sampled)
+        ))
+
+        valid_agent_state = [x[0] for x in valid_agents]
+        valid_agent_attrs = [x[1] for x in valid_agents]
+        valid_agent_rs = [x[2] for x in valid_agents]
+
+        predefined_agents_slice = slice(len(conditional_agent_states),len(conditional_agent_states)+len(region_predefined_agent_states))
+        regions[i].agent_states = response.agent_states[predefined_agents_slice] + valid_agent_state
+        regions[i].agent_attributes = response.agent_attributes[predefined_agents_slice] + valid_agent_attrs
+        regions[i].recurrent_states = response.recurrent_states[predefined_agents_slice] + valid_agent_rs
```

Comment:
Modifying Region objects in place is another thing that we need to be careful about in the documentation.

Reply:
Fixed so that it uses the member function `region[i].insert_all_agent_details` instead of modifying the fields directly.